app/helpers/authentication_helper.rb

```diff
@@ -23,6 +23,16 @@ def authentication_enabled_providers_for_user(user = current_user)
     Authentication::Providers.enabled_for_user(user)
   end
 
+  def signed_up_with(user = current_user)
+    providers = Authentication::Providers.enabled_for_user(user)
+
+    # If the user did not authenticate with any provider, they signed up with an email.
+    auth_method = providers.presence ? providers.map(&:official_name).to_sentence : "Email & Password"
```

Comment:
Nitpick: `present?` is probably more appropriate than `presence` here, but IMHO `any?` takes the cake for also being short:

```suggestion
    auth_method = providers.any? ? providers.map(&:official_name).to_sentence : "Email & Password"
```